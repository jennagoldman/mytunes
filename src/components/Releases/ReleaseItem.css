.releaseDiv {
  width: 300px;
}

.releaseDiv img {
  width: 250px;
}