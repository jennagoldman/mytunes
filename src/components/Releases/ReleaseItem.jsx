import React from 'react';
import PropTypes from 'prop-types';
import styles from './ReleaseItem.css';

const ReleaseItem = ({ releaseTitle, releaseId, coverArt }) => (
  <div className={styles.releaseDiv}>
    <h3>{releaseTitle}</h3>
    <img src={ coverArt ? `http://coverartarchive.org/release/${releaseId}/front` : 'https://www.placecage.com/g/250/250'} alt={releaseTitle} />
  </div>
);


ReleaseItem.propTypes = {
  releaseTitle: PropTypes.string.isRequired,
  releaseId: PropTypes.string.isRequired,
  coverArt: PropTypes.bool
};


export default ReleaseItem;
