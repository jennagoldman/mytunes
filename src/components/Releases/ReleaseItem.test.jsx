import React from 'react';
import { shallow } from 'enzyme';
import ReleaseItem from './ReleaseItem.jsx';

describe('ReleaseItem', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<ReleaseItem 
      releaseTitle={'Out of the Blue'} 
      releaseId={'123abc'} 
      coverArt={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
