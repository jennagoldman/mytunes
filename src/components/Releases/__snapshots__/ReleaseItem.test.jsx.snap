// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseItem matches a snapshot 1`] = `
<div
  className="releaseDiv"
>
  <h3>
    Out of the Blue
  </h3>
  <img
    alt="Out of the Blue"
    src="http://coverartarchive.org/release/123abc/front"
  />
</div>
`;
